proxy/src/test/java/com/wavefront/agent/PushAgentTest.java

```diff
@@ -503,4 +516,70 @@ private void gzippedHttpPost(String postUrl, String payload) throws Exception {
     connection.getOutputStream().flush();
     logger.info("HTTP response code (gzipped content): " + connection.getResponseCode());
   }
+
+
+  @Test
+  public void testDeltaCounterHandlerMixedData() throws Exception {
+    reset(mockSenderTask);
+    Capture<String> capturedArgument = Capture.newInstance(CaptureType.ALL);
+    mockSenderTask.add(EasyMock.capture(capturedArgument));
+    expectLastCall().atLeastOnce();
+    replay(mockSenderTask);
+
+    Socket socket = SocketFactory.getDefault().createSocket("localhost", deltaPort);
+    BufferedOutputStream stream = new BufferedOutputStream(socket.getOutputStream());
+    String payloadStr1 = "∆test.mixed1 1.0 source=test1\n";
+    String payloadStr2 = "∆test.mixed2 2.0 source=test1\n";
+    String payloadStr3 = "test.mixed3 3.0 source=test1\n";
+    String payloadStr4 = "∆test.mixed3 3.0 source=test1\n";
+    stream.write(payloadStr1.getBytes());
+    stream.write(payloadStr2.getBytes());
+    stream.write(payloadStr3.getBytes());
+    stream.write(payloadStr4.getBytes());
+    stream.flush();
+    TimeUnit.MILLISECONDS.sleep(10000);
+    socket.close();
+    verify(mockSenderTask);
+    String[] reportPoints = {"1.0", "2.0", "3.0"};
+    int pointInd = 0;
+    for (String s : capturedArgument.getValues()) {
+      Assert.assertTrue(s.startsWith("\"∆test.mixed" + Integer.toString(pointInd+1) + "\" " + reportPoints[pointInd]));
+      pointInd += 1;
+    }
+  }
+
+
+  @Test
+  public void testDeltaCounterHandlerDataStream() throws Exception {
+    reset(mockSenderTask);
+    Capture<String> capturedArgument = Capture.newInstance(CaptureType.ALL);
+    mockSenderTask.add(EasyMock.capture(capturedArgument));
+    expectLastCall().atLeastOnce();
+    replay(mockSenderTask);
+
+    Socket socket = SocketFactory.getDefault().createSocket("localhost", deltaPort);
+    BufferedOutputStream stream = new BufferedOutputStream(socket.getOutputStream());
+    String payloadStr = "∆test.mixed 1.0 " + startTime + " source=test1\n";
+    stream.write(payloadStr.getBytes());
+    stream.write(payloadStr.getBytes());
+    stream.flush();
+    TimeUnit.MILLISECONDS.sleep(6000);
+    stream.write(payloadStr.getBytes());
+    stream.flush();
+    TimeUnit.MILLISECONDS.sleep(1000);
+    stream.write(payloadStr.getBytes());
+    stream.write(payloadStr.getBytes());
+    stream.flush();
+    TimeUnit.MILLISECONDS.sleep(6000);
```

Comment:
Suggestion: Since you are dealing with time-sensitive components, you can get around having to wait for several seconds by calling reportCache() manually; and/or by using a custom Ticker for Caffeine cache.